hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/lease/LeaseManager.java

```diff
@@ -45,8 +49,8 @@
 
   private final String name;
   private final long defaultTimeout;
-  private final Object monitor = new Object();
   private Map<T, Lease<T>> activeLeases;
+  private BlockingQueue<T> leaseKeyBlockingQueue;
```

Comment:
We should use Semaphore in this case; see https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Semaphore.html

Reply:
@szetszwo Thanks for review, I have done changes as per patch, but added extra release in shutdown() as possibility for concurrency issue. i.e.
1. shutdown --> disable MonitorThread (set running to false)
2. interrupt Monitor thread

For step 2, if MonitorThread waiting, then interrupt will work else ignored. So later moves to waiting for semaphore, then there is no way to exist thread.

So added extra "semaphore.release()" in shutdown to ensure exit from tryAcquire() wait in above case.